sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -418,6 +419,13 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
 
   override def toString: String = treeString
 
+  def toOriginString: String =
+    if (this.origin.callSite.isDefined && !this.isInstanceOf[BoundReference]) {
```

Comment:
`AttributeReference` which is created with expression parsing is transformed into `BoundReference` so we can't generate comment properly when the query is like as follows.

```
SELECT col1, col2 FROM TBL WHERE ...
```

In the example above, col1 and col2 is parsed into `BoundReference` finally.